src/txn/txn_ckpt.c

```diff
@@ -478,7 +488,12 @@ __wt_checkpoint_get_handles(WT_SESSION_IMPL *session, const char *cfg[])
      * Decide whether the tree needs to be included in the checkpoint and if so, acquire the
      * necessary locks.
      */
+    time_start = __wt_clock(session);
     WT_SAVE_DHANDLE(session, ret = __checkpoint_lock_dirty_tree(session, true, force, true, cfg));
+    time_stop = __wt_clock(session);
+    time_diff = WT_CLOCKDIFF_US(time_stop, time_start);
+    ++S2C(session)->ckpt_lock;
+    S2C(session)->ckpt_lock_time += time_diff;
     WT_RET(ret);
```

Comment:
What about the call to `__wt_session_get_dhandle`? I reviewed the code and I think we cannot be blocked there but I may be wrong.

Reply:
If we want to save our time to debug this, can we also add stats inside `__checkpoint_lock_dirty_tree`?

Reply:
regarding timing stats for `__wt_session_get_dhandle ` inside the function `__wt_checkpoint_get_handles`, we are already gathering the stats for metadata check and lock tree. The remaining is the get dhandle. This timing stats should be recorded as part of the `checkpoint applied time`, so there is no need for specific stats to this.

Reply:
Added stats for `drop` also that is inside the `checkpoint_lock` function.